docs/sdks/uid2-sdk-ref-python.md

```diff
@@ -90,62 +97,101 @@ Available information returned through the SDK is outlined in the following tabl
+A UID2 sharer is any participant that wants to share UID2s with another participant. Raw UID2s must be encrypted into UID2 tokens before sending them to another participant. For an example of usage, see [examples/sample_sharing.py](https://github.com/IABTechLab/uid2-client-python/blob/master/examples/sample_sharing.py) script.
 
 The following instructions provide an example of how you can implement sharing using the UID2 SDK for Python, either as a sender or a receiver.
 
 1. Create a ```UID2Client``` reference:
  
    ```python
-   from uid2_client import Uid2Client
-   client = Uid2Client(base_url, auth_key, secret_key)
+   from uid2_client import Uid2ClientFactory
+   client = Uid2ClientFactory.create(base_url, auth_key, secret_key)
    ```
+   <!-- Alternative to the above for EUID:
+   ```python
+   from uid2_client import EuidClientFactory
+   client = EuidClientFactory.create(base_url, auth_key, secret_key) 
+   ``` -->
 2. Refresh once at startup, and then periodically (recommended refresh interval is hourly):
 
    ```python
-   keys = client.refresh_keys()
+   client.refresh_keys()
    ```
 
 3. Senders: 
    1. Call the `encrypt` function. Then, if encryption succeeded, send the UID2 token to the receiver:
 
       ```python
-      from uid2_client import encrypt
-      from uid2_client.identity_scope import IdentityScope
-      
       try:
-         identity_scope = IdentityScope.UID2
-         encrypted_data = encrypt(raw_uid, identity_scope, keys)
-         #send encrypted_data to receiver
+         encrypted_data = client.encrypt(raw_uid)
+         # send encrypted_data to receiver
       except EncryptionError as err:
-        #check for failure reason
+        # check for failure reason
         print(err)
-      ``` 
-<!-- Alternative to the above for EUID:
-      from uid2_client import encrypt
-      from uid2_client.identity_scope import IdentityScope
-      
-        try:
-         identity_scope = IdentityScope.UID2  # or IdentityScope.EUID
-         encrypted_data = encrypt(raw_uid, identity_scope, keys)
-         #send encrypted_data to receiver
-      except EncryptionError as err:
-        #check for failure reason
-        print(err) -->
+      ```
 
 4. Receivers:
    1. Call the `decrypt` function. Then, if decryption succeeded, use the raw UID2:
     
       ```python
-      from uid2_client import decrypt
       try:
-        result = decrypt(ad_token, keys)
-        #use successfully decrypted result.uid2
+        result = client.decrypt(ad_token)
+        # use successfully decrypted result.uid2
       except EncryptionError as err:
-        #check for failure reason
+        # check for failure reason
         print(err)
       ```
 
+## Usage for Publishers
+
+1. Create an instance of Uid2PublisherClient
```

Comment:
1. Create an instance of Uid2PublisherClient:
(add semicolon)

Reply:
added